### PR TITLE
Fixed rotation of gradients while rotating a VMobject

### PR DESCRIPTION
**Pull request by sparshg**
## Changelog / Overview

Fixed the direction of gradient which remained the same while rotating VMobjects
Added `rotate_sheen_direction()` method in VMobject

## Explanation for Changes

When VMobject is rotated, the direction of sheen is set correctly using a new method `rotate_sheen_direction()`. To call this I also added `rotate()` method in VMobject class
[Documentation build](https://manimce--1513.org.readthedocs.build/en/1513/reference/manim.mobject.types.vectorized_mobject.VMobject.html)

### Before
(Notice how you can replay it in a perfect loop which should not be the case)

https://user-images.githubusercontent.com/43041139/118327694-64873f00-b523-11eb-8d7f-f65af2f5a02a.mp4

### After

https://user-images.githubusercontent.com/43041139/118351255-84951d80-b578-11eb-80ce-4eea54e652f3.mp4

### Code
```py
class Test(Scene):
    def construct(self):
        c = VGroup(
            Circle(radius=1.5, fill_opacity=1).set_color(color=[WHITE, BLACK]),
            Circle(radius=1.5, fill_opacity=1).set_color(color=[WHITE, BLACK]),
            Square(fill_opacity=1).set_color(color=[GREEN, DARKER_GRAY]),
        ).arrange(buff=LARGE_BUFF)

        self.add(c)
        self.play(
            Rotating(c[0], OUT, PI),
            Rotating(c[1], UP, PI),
            Rotating(c[2], OUT, PI),
            run_time=2,
        )
```
## Testing Status

Pass (Linux/Manjaro)

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: manim/mobject/types/vectorized_mobject.py [hunk 1]
```diff
@@ -414,6 +414,16 @@ def set_sheen_direction(self, direction, family=True):
             self.sheen_direction = direction
         return self
 
+    def rotate_sheen_direction(self, angle, axis=OUT, family=True):
```
Comment (AntonBallmaier) [inline manim/mobject/types/vectorized_mobject.py:417]: You should add documentation to your new function. Also type hints would be a nice addition

File: manim/mobject/types/vectorized_mobject.py [hunk 2]
```diff
@@ -734,6 +744,17 @@ def apply_function(self, function):
...
 
+    def rotate(
+        self,
+        angle,
+        axis=OUT,
+        about_point: Optional[Sequence[float]] = None,
+        **kwargs,
+    ):
+        self.rotate_sheen_direction(angle, axis)
+        Mobject.rotate(self, angle, axis, about_point, **kwargs)
```
Comment (AntonBallmaier) [inline manim/mobject/types/vectorized_mobject.py:755]:
```suggestion
        super().rotate(angle, axis, about_point, **kwargs)
```

File: manim/mobject/types/vectorized_mobject.py [hunk 3]
```diff
@@ -734,6 +744,17 @@ def apply_function(self, function):
             self.make_smooth()
         return self
 
+    def rotate(
```
Comment (AntonBallmaier) [inline manim/mobject/types/vectorized_mobject.py:747]: Type hints again :)

**Review by AntonBallmaier — CHANGES_REQUESTED** [review]

File: manim/mobject/types/vectorized_mobject.py (hunk 1 above)
Comment (sparshg) [reply manim/mobject/types/vectorized_mobject.py]: So I added some docs and type hints, I am not sure if I did it correctly or not lol

**Comment by AntonBallmaier** [comment]
Docs reference: https://manimce--1513.org.readthedocs.build/en/1513/reference/manim.mobject.types.vectorized_mobject.VMobject.html

**Review by AntonBallmaier — APPROVED** [review]
Lgtm :)

**Comment by behackl** [comment]
@AntonBallmaier please make sure when merging PRs that they are labeled appropriately and that the PR title is sufficient for the changelog.

(The title is okay here, the labelling isn't. :-))

**Comment by AntonBallmaier** [comment]
> @AntonBallmaier please make sure when merging PRs that they are labeled appropriately and that the PR title is sufficient for the changelog.
> 
> (The title is okay here, the labelling isn't. :-))

Ok, I'll keep that in mind :)

File: manim/mobject/types/vectorized_mobject.py [hunk 4]
```diff
@@ -414,7 +433,57 @@ def set_sheen_direction(self, direction, family=True):
...
+            The extent of lustre/gradient to apply. If negative, the gradient
+            starts from black, if positive the gradient starts from white and
+            changes to the current color.
+        direction : :class:`numpy.ndarray`, optional
+            Direction from where the gradient is applied.
+
+        Examples
+        --------
+        .. manim:: SetSheen
+            :save_last_frame:
```
Comment (behackl) [inline manim/mobject/types/vectorized_mobject.py:480]: The example currently doesn't render correctly due to a missing newline before the code block.

```suggestion
            :save_last_frame:

```

**Review by behackl — CHANGES_REQUESTED** [review]

**Review by behackl — APPROVED** [review]

**Pull request merged**